File: app/src/main/java/ru/ulstu_team/ulstuschedule/ui/base/BaseFragment.java
package ru.ulstu_team.ulstuschedule.ui.base;

import android.app.Fragment;
import android.content.Context;

import ru.ulstu_team.ulstuschedule.App;
import ru.ulstu_team.ulstuschedule.injection.component.DaggerStudentScheduleComponent;
import ru.ulstu_team.ulstuschedule.injection.component.DaggerTeacherScheduleComponent;
import ru.ulstu_team.ulstuschedule.injection.component.StudentScheduleComponent;
import ru.ulstu_team.ulstuschedule.injection.component.TeacherScheduleComponent;
import ru.ulstu_team.ulstuschedule.injection.module.StudentScheduleModule;
import ru.ulstu_team.ulstuschedule.injection.module.TeacherScheduleModule;
import uk.co.chrisjenx.calligraphy.CalligraphyContextWrapper;

public abstract class BaseFragment extends Fragment {

    private TeacherScheduleComponent teacherScheduleComponent;
    private StudentScheduleComponent studentScheduleComponent;

    public TeacherScheduleComponent getTeacherScheduleComponent() {
        if (teacherScheduleComponent == null) {
            teacherScheduleComponent = DaggerTeacherScheduleComponent.builder()
                    .teacherScheduleModule(new TeacherScheduleModule(this))
                    .applicationComponent(App.get(this.getActivity()).getComponent())
                    .build();
        }
        return teacherScheduleComponent;
    }

    public StudentScheduleComponent getStudentScheduleComponent() {
        if (studentScheduleComponent == null) {
            studentScheduleComponent = DaggerStudentScheduleComponent.builder()
                    .studentScheduleModule(new StudentScheduleModule(this))
                    .applicationComponent(App.get(this.getActivity()).getComponent())
                    .build();
        }
        return studentScheduleComponent;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(CalligraphyContextWrapper.wrap(context));
    }
}

File: app/src/main/java/ru/ulstu_team/ulstuschedule/ui/base/BaseFragment.kt
package ru.ulstu_team.ulstuschedule.ui.base

import android.app.Fragment
import android.content.Context
import android.support.annotation.LayoutRes
import android.view.ViewStub

import ru.ulstu_team.ulstuschedule.App
import ru.ulstu_team.ulstuschedule.R
import ru.ulstu_team.ulstuschedule.injection.component.DaggerStudentScheduleComponent
import ru.ulstu_team.ulstuschedule.injection.component.DaggerTeacherScheduleComponent
import ru.ulstu_team.ulstuschedule.injection.component.StudentScheduleComponent
import ru.ulstu_team.ulstuschedule.injection.component.TeacherScheduleComponent
import ru.ulstu_team.ulstuschedule.injection.module.StudentScheduleModule
import ru.ulstu_team.ulstuschedule.injection.module.TeacherScheduleModule
import uk.co.chrisjenx.calligraphy.CalligraphyContextWrapper

abstract class BaseFragment : Fragment() {

    private var teacherScheduleComponent: TeacherScheduleComponent? = null
    private var studentScheduleComponent: StudentScheduleComponent? = null

    fun getTeacherScheduleComponent(): TeacherScheduleComponent {
        if (teacherScheduleComponent == null) {
            teacherScheduleComponent = DaggerTeacherScheduleComponent.builder()
                    .teacherScheduleModule(TeacherScheduleModule(this))
                    .applicationComponent(App[this.activity].component)
                    .build()
        }
        return teacherScheduleComponent!!
    }

    fun getStudentScheduleComponent(): StudentScheduleComponent {
        if (studentScheduleComponent == null) {
            studentScheduleComponent = DaggerStudentScheduleComponent.builder()
                    .studentScheduleModule(StudentScheduleModule(this))
                    .applicationComponent(App[this.activity].component)
                    .build()
        }
        return studentScheduleComponent!!
    }

    override fun onAttach(context: Context) {
        super.onAttach(CalligraphyContextWrapper.wrap(context))
    }
}
